src/runner.py

```diff
@@ -99,34 +131,39 @@ def run(self):
         try:
             while True:
                 checkout_node = self._db.receive_node(sub_id)
-
-                self._logger.log_message(logging.INFO, "Tarball: {}".format(
-                    checkout_node['artifacts']['tarball']
-                ))
-
-                self._logger.log_message(logging.INFO, "Creating test node")
-                node = self._create_node(checkout_node)
-
-                tmp = tempfile.TemporaryDirectory(dir=self._output)
-                output_file = self._generate_job(node, tmp.name)
-
-                self._logger.log_message(logging.INFO, "Running test")
-                process = self._runtime.submit(output_file, get_process=True)
+                process, tmp = self._schedule_test(checkout_node)
                 self._job_tmp_dirs[process] = tmp
-
                 self._cleanup_paths()
         except KeyboardInterrupt as e:
             self._logger.log_message(logging.INFO, "Stopping.")
         finally:
             self._db.unsubscribe(sub_id)
             self._cleanup_paths()
 
+    def run(self):
+        if self._node_id:
+            checkout_node = self._db.get_node(node_id)
```

Comment:
We need to check if node exists before running test to avoid errors like:
```File "/home/kernelci/./pipeline/runner.py", line 45, in _create_node
    'parent': checkout_node['_id'],
TypeError: 'NoneType' object is not subscriptable'
```

```suggestion
            checkout_node = self._db.get_node(self._node_id)
            if not checkout_node:
                self._logger.log_message(logging.ERROR,
                                         f"Node not found with id:\
{self._node_id}")
                return
            self._run_single_job(checkout_node)
```

Reply:
OK, however this is just one case of errors. The id could be a node that's not a checkout, or there could be an issue while communicating with the API. So I think it's better to handle it via exceptions.